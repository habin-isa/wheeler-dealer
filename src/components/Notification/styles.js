import styled from 'styled-components';

export const Wrapper = styled.div`
  position: absolute;
  background: white;
  z-index: 99999;
  width: max-content;
  text-align: left;
  padding: 10px;
  opacity: 0.9;
`;

export const List = styled.li`
  list-style-type: none;
  cursor: pointer;
  margin: 5px;
  font-size: 14px;
  margin-left: 10px;
`;