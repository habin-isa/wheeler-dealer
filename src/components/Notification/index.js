import React from 'react';
import * as S from './styles';
import { func, array } from 'prop-types';

const Notification = ({ likedList, deleteLike }) => {

  const likedNotifications = likedList.map((likedItem, i) => (
    <S.List
      key={i} 
      onClick={() => {
        deleteLike(i);
      }}
    >
      â¤¬ New like: {likedItem}
    </S.List>
  ));
  
  return (
    <S.Wrapper>{likedNotifications}</S.Wrapper>
  )
};

Notification.propTypes = {
  likedList: array,
  deleteLike: func
};

export default Notification;