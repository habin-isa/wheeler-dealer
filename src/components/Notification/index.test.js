
import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Notification from '.';

describe('Notification', () => {

  const props = {
    likedList: [
      'Man City shirt',
      'Buberry slippers',
      'Rapha top'
    ]
  };

  it('should render', () => {
    const { container } = render(<Notification likedList={props.likedList} />);
    expect(container.firstChild).toMatchSnapshot();
  });

   it('renders with correct like data', () => {
      shallow(<Notification likedList={props.likedList} />);
    });

});