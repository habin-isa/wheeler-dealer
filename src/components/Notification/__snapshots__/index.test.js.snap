// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render 1`] = `
<div
  class="sc-AxjAm cjYHtl"
>
  <li
    class="sc-AxirZ lgUHZh"
  >
    ⤬ New like: 
    Man City shirt
  </li>
  <li
    class="sc-AxirZ lgUHZh"
  >
    ⤬ New like: 
    Buberry slippers
  </li>
  <li
    class="sc-AxirZ lgUHZh"
  >
    ⤬ New like: 
    Rapha top
  </li>
</div>
`;
